### ISSUE TITLE
write function write_lab_results_to_results_project

### ISSUE THREAD

**Issue by pbchase** [ctsit/tt_engine#10]
write the function `write_lab_results_to_results_project`. this function should fetch its inputs from `get_lab_result_data` (name the dataframe `lab_result_data`) and results project. 

From the results project the function will read from the _UF Identifiers and Determination_ form and the _Subject Information and Results_ form. The latter form is a repeating form. It will need to read these fields:

```
record_id (aka UFID)
instance_number
redcap_repeating_instrument_name (or some such)
acc_number (to be added)
```

Name this second dataset, `lab_result_rows_in_redcap`. 

1. **Identify lab data with bad UFIDs** Create a data frame of distinct(lab_result_rows_in_redcap.record_id). Anti-join this to  by `lab_result_data.ufid == (those distinct record ids).record_id` to get records in the former that are not in the latter. Create a dataset `lab_records_with_unknown_ufids` from that anti-join. That will be appended as a sheet in an excel file that will be emailed to us. 

2. **Identify new lab results** left join the two inputs by `get_lab_result_data.ufid == lab_result_rows_in_redcap.record_id` and `acc_number`. Set appropriate instance numbers and instrument _name for the records that do not have them and name the data frame `new_lab_results`. Load `new_lab_results` records into the results project.

I'll add more features to this function in a new issue

**Comment by pbchase** [comment]
Also read `cleared_for_campus` from the redcap results project

**Comment by pbchase** [comment]
See https://drive.google.com/open?id=1-VjOGtVozgoejUweM0awuqk3LlDHAlrS

**Comment by ChemiKyle** [comment]
I think there must be something wrong with one of the datasources. Less than 10% of the UFIDs from `get_lab_results_data` match to UFIDs provided by `get_person_data`

```r
anti_join(lab_result_data, ufids_in_redcap, by = c('uf_id_from_epic' = 'record_id')) %>% nrow()
# 1030
inner_join(lab_result_data, ufids_in_redcap, by = c('uf_id_from_epic' = 'record_id')) %>% nrow()
# 104
```

**Comment by pbchase** [comment]
the get_person_data dataset is only 1100 records. I don't believe it is representative of the people who got tested.  @senrabc and I were lobbying to get all 70k records today, but we have yet to prevail. We think they will give us 20K - 30K.

**Comment by pbchase** [comment]
All that said, just put the 90% failed records into the `lab_records_with_unknown_ufids` dataset. We will follow the plan and report those to the person hub managers and it will be their responsibility to fix it.

**Comment by ChemiKyle** [comment]
Addressed by #14 and #18, further problems with this function are covered by other issues so I am closing this issue.
